nucleus/common/glassfish-api/src/main/java/org/glassfish/api/admin/AdminCommandLock.java

```diff
@@ -507,55 +505,57 @@ public void run() {
                     else
                         suspendCommandsTimedOut = true;
                 } catch (java.lang.InterruptedException e) {
-                    logger.log(Level.FINE, 
-                               "Interrupted acquiring command lock. ",
-                               e); 
-               }
+                    logger.log(Level.FINE, "Interrupted acquiring command lock. ", e);
+                }
             }
+            try {
```

Comment:
The body is doing various things left and right and it has two flags saying the same thing in a different way.
There is  the `lockAcquired` variable and the `suspendCommandsTimedOut` field (which means lock was not acquired because of a timeout trying). Because it is a field its unclear if or how this can be simplified. Therefore, I think, a `try` block has to put around the body as soon as the lock is acquired that does `unlock` in its `finally` at the end only in case `lockAcquired` is true.